composer/trainer/trainer.py

```diff
@@ -36,6 +36,7 @@
 from composer.devices import Device, DeviceCPU, DeviceGPU, DeviceMPS, DeviceTPU
 from composer.loggers import (ConsoleLogger, Logger, LoggerDestination, ProgressBarLogger, RemoteUploaderDownloader,
                               WandBLogger)
+from composer.metrics import METRIC_DEFAULT_CTORS
```

Comment:
I'll admit I'm a bit confused why we need to modify `trainer.py`... do default metrics make sense for Composer? I'm not sure why we would want 'InContextLearningLMAccuracy' if training diffusion models, for instance.

Reply:
My main concern was the awkwardness of having the user specify the metric in two places (in the eval data loader and in the model) and my concern about whether that would cause issues if it were always specified in the model but only optionally instantiated with the eval data loader. Though upon second glance at the code I'm not sure if that latter concern holds.

Reply:
yeah, without a pretty compelling reason, I would avoid putting in this default metric constructor thing. The intention is that models define the instantiation of their metrics, and evaluators specify which of those metrics to use. If there is an issue with this design, we should def revisit, but probably don't want to change the design as part of this PR.